### PR TITLE
Github Actions: Send a "Failure" message to Slack on any SecRel failure

### PR DESCRIPTION
**Pull request by sethdarragile6**
## What was the problem?
Failed SecRel scans were only being notified to one dedicated email address.

Associated tickets or Slack threads:
- #927 

## How does this fix it?[^1]
A failed SecRel scan will send a [slack message](https://dsva.slack.com/archives/C04CA47HV96/p1675298668567209) to the #benefits-vro-devops channel

## How to test this PR
- Introduce a known vulnerability into code, push it and mirror it to abd-vro-internal
- A message is sent to #benefits-vro-devops, detailing the branch, originator of the change and a link to the failed job.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: .github/workflows/secrel.yml [hunk 1]
```diff
@@ -18,6 +18,11 @@ env:
   # while secrets.ACCESS_TOKEN has `repo` and `write:packages` permission
   GITHUB_ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN_GRADLE_BUILD }}
 
+  # Id for the #benefits-vro-devops Slack channel
+  SLACK_CHANNEL: C04CA47HV96
+  DEPLOY_STATE_EMOJI: 'x'
+  JOB_STATUS: 'failed'
```
Comment (ahay-agile6) [inline .github/workflows/secrel.yml:24]: do we need these two env vars?
```
DEPLOY_STATE_EMOJI: 'x'
JOB_STATUS: 'failed'
```
it looks like they're not used right now!

Comment (ahay-agile6) [reply .github/workflows/secrel.yml:24]: unless they're used in another job that's not in this file...

**Review by ahay-agile6 — APPROVED** [review]
lgtm!
just one question :)

File: .github/workflows/secrel.yml (hunk 1 above)
Comment (sethdarragile6) [reply .github/workflows/secrel.yml]: They are not! Good call, removed now.

**Review by advcoder — APPROVED** [review]

File: .github/workflows/secrel4.yml [hunk 2]
```diff
@@ -107,3 +110,21 @@ jobs:
       config-file: .github/secrel/config4.yml
       images: ${{ needs.images-for-secrel.outputs.all-images }}
     secrets: inherit
+
+  notify-secrel-error:
+    if: failure()
+    needs: [publish-images, images-for-secrel, secrel]
```
Comment (yoomlam) [inline .github/workflows/secrel4.yml:116]: `needs: secrel`
This step is notifying about SecRel failures and doesn't use output from other steps.

File: .github/workflows/secrel4.yml [hunk 3]
```diff
@@ -107,3 +110,21 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Notify Slack on SecRel Failure"
+        id: notify-slack
```
Comment (yoomlam) [inline .github/workflows/secrel4.yml:121]: Don't need an `id` since it's not referenced.

**Pull request merged**